openerp/addons/base/migrations/9.0.1.3/pre-migration.py

```diff
@@ -54,7 +54,23 @@
 }
 
 
+OBSOLETE_RULES = (
+    'multi_company_default_rule',
+    'res_currency_rule',
+)
+
+
+def remove_obsolete(cr):
+    openupgrade.logged_query(cr, """
+        delete from ir_rule rr
+        using ir_model_data d where rr.id=d.res_id
+        and d.model = 'ir.rule' and d.module = 'base'
+        and d.name in {}
+        """.format(OBSOLETE_RULES))
```

Comment:
@hbrunn Every change was done because the database would fail to run after migration. In this case, model multi_company.default is gone from 9.0. I'm happy to see I'm not the only one surprised they weren't cleaned up.